faker/providers/address/it_IT/__init__.py

```diff
@@ -511,11 +17616,19 @@ class Provider(AddressProvider):
         "{{street_suffix}} {{last_name}}",
     )
     street_address_formats = (
-        "{{street_name}} {{building_number}}",
-        "{{street_name}} {{building_number}} {{secondary_address}}",
+        "{{street_name}}, {{building_number}}",
+        "{{street_name}}, {{building_number}} {{secondary_address}}",
     )
-    address_formats = ("{{street_address}}\n{{city}}, {{postcode}} {{state}} ({{state_abbr}})",)
-    secondary_address_formats = ("Appartamento ##", "Piano #")
+    address_formats = ("{{street_address}}\n{{postcode_city_province}}",)
+    secondary_address_formats = ("Appartamento @#", "Piano #")
+
+    def postcode_city_province(self) -> str:
+        cap = self.postcode()
+        rand_city_prov = self.random_element(self.cap_city_province[cap])  # type: List[str]
+        return cap + ", " + rand_city_prov[0] + " (" + rand_city_prov[1] + ")"
+
+    def city(self):
```

Comment:
can you add a type hint here too?

```suggestion
    def city(self) -> str:
```